src/cloudant/_common_util.py

```diff
@@ -263,6 +263,16 @@ def get_docs(r_session, url, encoder=None, headers=None, **params):
     resp.raise_for_status()
     return resp
 
+#pylint: disable=unused-argument
+def append_response_error_content(response, **kwargs):
+    """
+    Provides a helper to add HTTP response error and reason messages.
+    """
+    if response.status_code >= 400:
+        resp_content = response.json()
+        response.url += ' %s %s' % (resp_content['error'], resp_content['reason'])
```

Comment:
Added check to prevent raising errors with `response.json()`.  Also added checks to verify that `error` and `reason` keys exist in 778b332.